ramanujan/cmf/cmf.py

```diff
@@ -13,7 +13,8 @@ class CMF:
     $Mx(x, y) \cdot My(x+1, y) = My(x, y) \cdot Mx(x, y+1)$
     """
 
-    def __init__(self, Mx: Matrix, My: Matrix):
+    def __init__(self, Mx: Matrix, My: Matrix, maximal_cache_dims=(50, 50), 
+        initial_mat=Matrix(sp.eye(2)), initial_loc=(1,1), potential_cache_file=None):
```

Comment:
There are a lot of new parameter, which are only required for the caching, and not the "standard" CMF construction. Consider putting them into a single object, so it won't be confusing